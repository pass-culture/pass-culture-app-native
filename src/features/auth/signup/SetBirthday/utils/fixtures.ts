import { dateDiffInFullYears } from 'libs/dates'

export const CURRENT_DATE = new Date('2020-12-01T00:00:00.000Z')
export const ELIGIBLE_AGE_DATE = new Date('2003-12-01T00:00:00.000Z')
export const DEFAULT_SELECTED_DATE = new Date('2006-12-01T00:00:00.000Z')
export const MINIMUM_DATE = new Date('1900-01-01T00:00:00.000Z')
export const MAXIMUM_DATE = new Date('2003-12-01T00:00:00.000Z')
export const FUTUR_DATE = new Date('2022-12-01T00:00:00.000Z')
export const FIFTEEN_YEARS_OLD_FIRST_DAY_DATE = new Date('2005-12-01T00:00:00.000Z')
export const NOT_ELIGIBLE_YOUNGEST_AGE_DATE = new Date('2006-01-01T00:00:00.000Z')
export const AGE = dateDiffInFullYears(new Date(NOT_ELIGIBLE_YOUNGEST_AGE_DATE), CURRENT_DATE)
