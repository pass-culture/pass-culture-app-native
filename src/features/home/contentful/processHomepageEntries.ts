import {
  HomepageEntries,
  Module,
  ModuleFields,
  CONTENT_TYPES,
  CoverParameters,
  AlgoliaParameters,
} from './contentful.d'
import {
  Offers,
  OffersWithCover,
  ExclusivityPane,
  BusinessPane,
  ProcessedModule,
} from './moduleTypes'

export const processHomepageEntries = (homepage: HomepageEntries): Array<ProcessedModule> => {
  const {
    fields: { modules },
  } = homepage

  const processedModules = modules.map((module: Module) => {
    const { fields }: { fields: ModuleFields } = module
    if (!fields || !hasAtLeastOneField(fields)) return

    const contentType = getContentType(module)
    if (contentType === CONTENT_TYPES.ALGOLIA) {
      const { algoliaParameters, displayParameters, cover } = fields

      // algoliaParameters and displayParameters are required for algolia modules
      if (!algoliaParameters || !displayParameters || !hasAtLeastOneField(algoliaParameters)) return
      const { fields: algolia } = algoliaParameters
      const { fields: display } = displayParameters

      if (cover && hasAtLeastOneField(cover)) {
        return new OffersWithCover({ algolia, cover: buildImageUrl(cover.fields), display })
      } else {
        return new Offers({ algolia, display })
      }
    }

    if (contentType === CONTENT_TYPES.EXCLUSIVITY) {
      const { alt, offerId } = fields
      const image = buildImageUrl(fields)
      // Those 3 fields are required for the exclusivity module in Contentful
      if (alt && image && offerId) return new ExclusivityPane({ alt, image, offerId })
    }

    if (contentType === CONTENT_TYPES.BUSINESS) {
      const { firstLine, secondLine, url } = fields
      return new BusinessPane({ firstLine, image: buildImageUrl(fields), secondLine, url })
    }

    return
  })

  // We shouldn't be needing the 'as' next line
  return processedModules.filter(Boolean) as Array<ProcessedModule>
}

const buildImageUrl = (fields: CoverParameters['fields']): string | null => {
  const image = fields.image
  if (image && hasAtLeastOneField(image.fields)) {
    const { url } = image.fields.file
    return url ? `https:${url}` : null
  }
  return null
}

const hasAtLeastOneField = (
  object: ModuleFields | CoverParameters | AlgoliaParameters | Record<string, unknown>
) => {
  return Object.keys(object).length > 0
}

const getContentType = (module: Module) => {
  const { contentType } = module.sys
  return contentType && contentType.sys.id
}
