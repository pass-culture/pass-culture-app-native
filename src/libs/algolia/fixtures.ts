import { LocationType } from 'features/search/enums'
import { SearchQueryParameters } from 'libs/algolia/types'

export const SearchQueryParametersFixture: SearchQueryParameters = {
  beginningDatetime: undefined,
  date: null,
  endingDatetime: undefined,
  hitsPerPage: 20,
  locationFilter: { locationType: LocationType.EVERYWHERE },
  offerCategories: [],
  offerSubcategories: [],
  offerIsDuo: false,
  offerIsFree: false,
  offerIsNew: false,
  offerTypes: {
    isDigital: false,
    isEvent: false,
    isThing: false,
  },
  priceRange: null,
  query: '',
  tags: [],
  timeRange: null,
  page: 0,
}
