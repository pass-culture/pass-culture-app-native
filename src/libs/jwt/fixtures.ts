export const tokenRemainingLifetimeInMs = 10 * 60 * 1000
const defaultDecodedToken = {
  iat: 1691670780,
  jti: '7f82c8b0-6222-42be-b913-cdf53958f17d',
  sub: 'bene_18@example.com',
  nbf: 1691670780,
  user_claims: { user_id: 1234 },
}
export const decodedTokenWithRemainingLifetime = {
  ...defaultDecodedToken,
  exp: (Date.now() + tokenRemainingLifetimeInMs) / 1000,
}
