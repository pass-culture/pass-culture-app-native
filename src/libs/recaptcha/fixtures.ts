export const UnknownErrorFixture =
  '{ "message": "error", "errorCode": "UnknownError", "error": "someError" }'

export const NetworkErrorFixture = `{ "message": "error", "errorCode": "ReCaptchaNetworkError" }`
