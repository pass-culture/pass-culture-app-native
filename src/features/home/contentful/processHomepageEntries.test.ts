import { adaptedHomepageEntries } from 'tests/fixtures/homepageEntries'

import { processHomepageEntries } from './processHomepageEntries'

describe('processHomepageEntries', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('should format homepage entries so we can use it to display infos on home page', () => {
    expect(processHomepageEntries(adaptedHomepageEntries)).toMatchSnapshot()
  })

  it('should format homepage entries: no fields case', () => {
    const sys = {
      space: { sys: { type: 'Link', linkType: 'Space', id: '2bg01iqy0isv' } },
      id: '16PgpnlCOYYIhUTclR0oO4',
      type: 'Entry',
      createdAt: '2020-07-02T13:36:02.919Z',
      updatedAt: '2020-10-30T10:07:29.549Z',
      environment: { sys: { id: 'testing', type: 'Link', linkType: 'Environment' } },
      revision: 154,
      contentType: { sys: { type: 'Link', linkType: 'ContentType', id: 'homepage' } },
      locale: 'en-US',
    }

    const emptyModulesHomepageEntries = {
      sys,
      fields: {
        title: 'Homepage',
        modules: [],
      },
    }
    expect(processHomepageEntries(emptyModulesHomepageEntries)).toEqual([])
  })
})
