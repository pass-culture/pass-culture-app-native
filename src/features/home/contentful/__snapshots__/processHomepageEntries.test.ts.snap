// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processHomepageEntries should format homepage entries so we can use it to display infos on home page 1`] = `
Array [
  ExclusivityPane {
    "alt": "Exclusitivité du moment",
    "image": "https://images.ctfassets.net/2bg01iqy0isv/2qTXOFUocq1HhgB7Wzl23K/a48ab7bedde13545231c0f843fbcfe10/image__3_.png",
    "moduleId": "KiG6rWuQYhHuGIBXZNeB2",
    "offerId": "AE67",
  },
  Offers {
    "algolia": Object {
      "hitsPerPage": 10,
      "tags": Array [
        "	starter_pack_rentree",
      ],
      "title": "Pour bien commencer...",
    },
    "display": Object {
      "layout": "one-item-medium",
      "minOffers": 2,
      "title": "Pour bien commencer...",
    },
    "moduleId": "GlQb8Zg3n1PoYVmOyVCgW",
  },
  Offers {
    "algolia": Object {
      "hitsPerPage": 4,
      "tags": Array [
        "visites",
      ],
      "title": "Puisque tu aimes les visites",
    },
    "display": Object {
      "layout": "two-items",
      "minOffers": 2,
      "title": "Puisque tu aimes les visites...",
    },
    "moduleId": "5u3NaFTcu2XR5KgJ1OAg8C",
  },
  Offers {
    "algolia": Object {
      "hitsPerPage": 4,
      "tags": Array [
        "offre_du_moment",
      ],
      "title": "Les offres du moment",
    },
    "display": Object {
      "layout": "one-item-medium",
      "minOffers": 1,
      "title": "Offres du moment",
    },
    "moduleId": "1LgDMVOKdH3agA0WG2LFEr",
  },
  Offers {
    "algolia": Object {
      "beginningDatetime": "2020-09-11T00:01+02:00",
      "endingDatetime": "2020-09-12T23:59+02:00",
      "hitsPerPage": 30,
      "title": "Week-end",
    },
    "display": Object {
      "layout": "two-items",
      "minOffers": 2,
      "title": "A faire ce week-end !",
    },
    "moduleId": "3AbfVZZIZhfvSfGxJSrSqe",
  },
  BusinessPane {
    "firstLine": "Donne ton avis...",
    "image": "https://images.ctfassets.net/2bg01iqy0isv/6kgALLZ7PL6vYubjvE8s0c/8e545d3312343d25c776c3cded9e2784/Capture_d___e__cran_2020-07-22_a___12.24.11.png",
    "moduleId": "K2yYJkeX30OoepIooo2Um",
    "secondLine": "En répondant à notre rapide questionnaire",
    "targetNotConnectedUsersOnly": undefined,
    "url": "https://passculture.typeform.com/to/PYl2WwKC?email={email}",
  },
  BusinessPane {
    "firstLine": "En savoir plus sur le pass Culture",
    "image": "https://images.ctfassets.net/2bg01iqy0isv/6kgALLZ7PL6vYubjvE8s0c/8e545d3312343d25c776c3cded9e2784/Capture_d___e__cran_2020-07-22_a___12.24.11.png",
    "moduleId": "24FUVnnPPJ9v7JHkO7eaXK",
    "secondLine": "Consulte notre FAQ !",
    "targetNotConnectedUsersOnly": true,
    "url": "https://aide.passculture.app/fr/",
  },
  BusinessPane {
    "firstLine": "Crée un compte !",
    "image": "https://images.ctfassets.net/2bg01iqy0isv/6kgALLZ7PL6vYubjvE8s0c/8e545d3312343d25c776c3cded9e2784/Capture_d___e__cran_2020-07-22_a___12.24.11.png",
    "moduleId": "24FUVnnPPJ9v7JHkO7eaID",
    "secondLine": "Si tu as 18 ans, bénéficie de 300 €.",
    "targetNotConnectedUsersOnly": false,
    "url": "https://aide.passculture.app/fr/",
  },
  Offers {
    "algolia": Object {
      "hitsPerPage": 10,
      "tags": Array [
        "duo",
      ],
      "title": "Offres duo",
    },
    "display": Object {
      "layout": "two-items",
      "minOffers": 2,
      "title": "A faire à deux",
    },
    "moduleId": "3HnwRT9WtAl8pxywz83sX3",
  },
  OffersWithCover {
    "algolia": Object {
      "hitsPerPage": 2,
      "tags": Array [
        "foot",
      ],
      "title": "Sélection Mbappé",
    },
    "cover": "https://images.ctfassets.net/2bg01iqy0isv/bRKhK9wmpNkD6l5Zeo71J/cc59c22fe0382b61e831e4693e32826f/1020_Mbapp__2.png",
    "display": Object {
      "layout": "one-item-medium",
      "minOffers": 2,
      "title": "La sélection de ...",
    },
    "moduleId": "2Fjxhdirgh17ZvdPNZGDP4",
  },
  Offers {
    "algolia": Object {
      "hitsPerPage": 4,
      "tags": Array [
        "goncourt",
      ],
      "title": "Goncourt",
    },
    "display": Object {
      "layout": "two-items",
      "minOffers": 2,
      "title": "Découvre les livres sélectionnés pour le Goncourt des lycéens 2020 !",
    },
    "moduleId": "7c7xq3ulzqO2yDRJLDmths",
  },
  Offers {
    "algolia": Object {
      "categories": Array [
        "Instruments de musique",
        "Musique",
      ],
      "hitsPerPage": 6,
      "title": "Musique",
    },
    "display": Object {
      "layout": "one-item-medium",
      "minOffers": 2,
      "title": "Parce que tu aimes la musique...",
    },
    "moduleId": "6lk6vCol5Qza2mfdtsTzW",
  },
]
`;
